MekHQ/src/mekhq/campaign/universe/RandomFactionGenerator.java

```diff
@@ -734,7 +735,9 @@ public String getMissionTarget(String attacker, String defender, Date date) {
 						if (f.getShortName().equals(defender) ||
 								defender.equals("PIR") ||
 								(f.getShortName().equals(attacker) && defender.equals("REB"))) {
-							planetList.add(planetKey);
+							if(filterByDistance && currentPlanet.getDistanceTo(planetKey) <= searchRadius) {
```

Comment:
This can be combined with the previous if statement rather than adding another level of nesting.

Reply:
I'll go ahead and make that change.